rfcs/RFC-0015-onion-services/0015-onion-services.md

```
@@ -0,0 +1,1095 @@
+session-data protocol
+([RFC-0008](../RFC-0008-session-protocol/0008-session-protocol.md)) inside the
+e2e-encrypted channel.
+
+### 4.2 Service identity, naming and delegation
+
+#### 4.2.1 Identity key and self-certifying address
+
+A service is identified by an Ed25519 [07] key pair `id_S = (sk_S, pk_S)`. The
+service's canonical **self-certifying address** is:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)